lib/AST/Mangle.cpp

```diff
@@ -1603,7 +1603,12 @@ void Mangler::mangleEntity(const ValueDecl *decl,
       return mangleAccessorEntity(accessorKind, func->getAddressorKind(),
                                   func->getAccessorStorageDecl());
   }
-  
+
+  // Avoid mangling nameless entity. This may happen in erroneous code as code
+  // completion.
+  if (!decl->hasName())
```

Comment:
This is to improve the mangler to allow nameless entities, in this test case `let:{`. As far as i know, we don't have a default name to set on the var decl lead by `let`, so to fix mangler to be more forgiving is a reasonable solution.